test/test_rosenbrock.py

```diff
@@ -0,0 +1,246 @@
+
+        fail = False
+        return funcs, fail
+
+    def sens(self, xdict, funcs):
+        self.ng += 1
+        x = xdict["xvars"]
+        funcsSens = {}
+        grads = np.zeros(len(x))
+
+        for i in range(len(x) - 1):
+            grads[i] += 2 * (200 * x[i] ** 3 - 200 * x[i] * x[i + 1] + x[i] - 1)
+            grads[i + 1] += 200 * (x[i + 1] - x[i] ** 2)
+
+        funcsSens["obj"] = {"xvars": grads}
+
+        fail = False
+        return funcsSens, fail
+
+    def optimize(self, optName, tol, optOptions={}, storeHistory=False, hotStart=None):
+        self.nf = 0  # number of function evaluations
+        self.ng = 0  # number of gradient evaluations
+        # Optimization Object
+
+        optProb = Optimization("Rosenbrock Problem", self.objfunc)
+
+        n = 4  # Number of design variables
+        np.random.seed(10)
+        value = np.random.normal(size=n)
+
+        lower = np.ones(n) * -50
+        upper = np.ones(n) * 50
+        optProb.addVarGroup("xvars", n, lower=lower, upper=upper, value=value)
+
+        # Objective
+        optProb.addObj("obj")
+
+        # Check optimization problem:
+        print(optProb)
+
+        # Optimizer
+        try:
+            opt = OPT(optName, options=optOptions)
+        except Error:
+            raise unittest.SkipTest("Optimizer not available:", optName)
+
+        # Solution
+        if storeHistory is not None:
+            if storeHistory is True:
+                self.histFileName = "%s_Rsbrk_Hist.hst" % (optName.lower())
+            elif isinstance(storeHistory, str):
+                self.histFileName = storeHistory
+        else:
+            self.histFileName = None
+
+        sol = opt(optProb, sens=self.sens, storeHistory=self.histFileName, hotStart=hotStart)
+
+        # Test printing solution to screen
+        print(sol)
+
+        # Check Solution
+        self.fStar1 = 0.0
+
+        self.xStar1 = np.ones(n)
+
+        dv = sol.getDVs()
+        sol_xvars = [sol.variables["xvars"][i].value for i in range(n)]
+
+        assert_allclose(sol_xvars, dv["xvars"], atol=tol, rtol=tol)
+
+        assert_allclose(sol.objectives["obj"].value, self.fStar1, atol=tol, rtol=tol)
+        assert_allclose(dv["xvars"], self.xStar1, atol=tol, rtol=tol)
+
+    def check_hist_file(self, optimizer, tol):
+        """
+        We check the history file here along with the API
+        """
+        hist = History(self.histFileName, flag="r")
+        # Metadata checks
+        metadata = hist.getMetadata()
+        self.assertEqual(metadata["optimizer"], optimizer)
+        metadata_def_keys = ["optName", "optOptions", "nprocs", "startTime", "endTime", "optTime", "version"]
+        for key in metadata_def_keys:
+            self.assertIn(key, metadata)
+        hist.getOptProb()
+
+        # Info checks
+        self.assertEqual(hist.getDVNames(), ["xvars"])
+        # self.assertEqual(hist.getConNames(), ["con"])
```

Comment:
Do we want to keep these lines as comments or remove them?

Reply:
These come from `test_hs_071` as well. I don't have any strong opinion about that , I left them as "placeholders" for a later refactoring, but we can take them off, up to you guys!

Reply:
@nwu63 Any thoughts on this?

Reply:
I think we should remove these